eloop.c

```diff
@@ -571,6 +591,11 @@ void eloop_wait_for_read_sock(int sock)
 	if (sock < 0)
 		return;
 
+	if (sock >= FD_SETSIZE) {
+		warn_too_many_fds();
```

Comment:
I'm okay with this being added, but a quick grep through the code doesn't seem to find any in-tree callers for eloop_wait_for_read_sock.

Maybe we should delete it instead?

Reply:
Yes, I only see `eloop_register_read_sock()`. and `eloop_unregister_read_sock()` used, but not `eloop_wait_for_read_sock()`. Also, it looks like `eloop_register_sock()`, `eloop_unregister_sock()`, and `os_get_time()` can all be made static.

Reply:
Actually, I found several things to clean-up in eloop.c/eloop.h including function prototypes in the header files that refer to functions that have never existed in the Git history. I'll save that clean-up for another PR, but I think I can drop that function for now.

Reply:
If you are okay with it, I'll merge this patch and backport it to the 1.1 branch.